core/domain/change_domain.py

```diff
@@ -133,7 +131,10 @@ class BaseChange:
         'deprecated_values': {}
     }]
 
-    def __init__(self, change_dict: Dict[str, ChangeDictValueTypes]) -> None:
+    # Here, Argument `change_dict` can accept dictionaries which have arbitrary
+    # number of keys with different types of values. So, to make the argument
+    # generalized for every type of dictionaries, we used Any type here.
+    def __init__(self, change_dict: Dict[str, Any]) -> None:
```

Comment:
Actually, I think there are different types of change dicts. So shouldn't we be doing a union type instead (of all the allowed ones)?

'Any' literally means every type is allowed, but in our case, we only admit a finite number of specific types.

Reply:
Here, I also used `Mapping` because:

`Dict[str, str]` can never be a subtype of `Dict[str, Union[str, int]]`, because MyPy considers both of them as different types of definitions. Due to the invariant property of `Dict`.

But `Dict[str, str]` can be a subtype of `Mapping[str, Union[str, int]]`. 

The only concern here is that `Mapping` does not contain some of the `Dict`'s method. But to encounter this issue we can annotate the return type of all the publically accesable methods as `Dict` ( already done in `to_dict()` method )